src/helpers/NRF52Board.cpp

```diff
@@ -0,0 +1,17 @@
+#if defined(NRF52_PLATFORM)

```

Comment:
I don't really like the preprocessor guards here. Better put the file into the `nrf52` subdir and use `build_src_filter` in `platformio.ini`.

Reply:
That makes sense.
Let me take a look.

Reply:
@fschrempf NRF52_PLATFORM is sure to work.
<img width="1128" height="542" alt="image" src="https://github.com/user-attachments/assets/bd4db74a-21c1-47c6-9bba-6205a026c265" />

If you want to use build_src_filter in platformio.ini, we need to add hepers/nrf52 in build_src_filter for 20 platformio.ini of 20 NRF52 variants.
Or we need to add into nrf52_base. This will have much larger impact.

<img width="988" height="573" alt="image" src="https://github.com/user-attachments/assets/4849751e-6764-4da7-850d-8e63a860789a" />

Reply:
As you can see in my PR, I simply added `helpers/nrf52/NRF52Board.cpp` to `nrf52_base`, which should be ok for now. The `SerialBLEInterface` is problematic because it depends on two conditions: NRF52 and BLE. This is difficult to reflect in `platformio.ini` cleanly without duplication (which I consider a major flaw in PlatformIO, also relevant in for this proposal [here](https://github.com/meshcore-dev/MeshCore/issues/758), but let's not take it too far).

Anyway, this is just my personal view on how to do it. Do take it with a grain of salt.